storage/src/main/java/tech/pegasys/teku/storage/store/StoreTransactionUpdates.java

```diff
@@ -210,17 +229,24 @@ public void applyToStore(final Store store) {
     tx.time.ifPresent(value -> store.time = value);
     tx.genesis_time.ifPresent(value -> store.genesis_time = value);
     tx.justified_checkpoint.ifPresent(value -> store.justified_checkpoint = value);
-    tx.finalized_checkpoint.ifPresent(value -> store.finalized_checkpoint = value);
+    newFinalizedCheckpoint.ifPresent(value -> store.finalized_checkpoint = value.getCheckpoint());
```

Comment:
For consistency, we're now only setting a new finalized checkpoint if the `tx` checkpoint made it through the epoch check (if the new finalized epoch is greater than the current finalized epoch): https://github.com/PegaSysEng/teku/blob/4963b9098e9f92d28dd72e58ae01f6023439b8e2/storage/src/main/java/tech/pegasys/teku/storage/store/StoreTransactionUpdates.java#L82-L84